Sources/MessagingPush/MessagingPushImplementation.swift

```diff
@@ -61,13 +66,19 @@ internal class MessagingPushImplementation: MessagingPushInstance {
             logger.info("no profile identified, so not registering device token to a profile")
             return
         }
-
-        _ = backgroundQueue.addTask(type: QueueTaskType.registerPushToken.rawValue,
-                                    data: RegisterPushNotificationQueueTaskData(profileIdentifier: identifier,
-                                                                                deviceToken: deviceToken,
-                                                                                lastUsed: Date()),
-                                    groupStart: .registeredPushToken(token: deviceToken),
-                                    blockingGroups: [.identifiedProfile(identifier: identifier)])
+        deviceAttributes(deviceToken: deviceToken) { attributes in
+            var deviceAttributes = attributes ?? customAttributes
+            if let customDeviceAttributes = customAttributes, let defaultDeviceAttributes = attributes {
+                deviceAttributes = defaultDeviceAttributes.merging(customDeviceAttributes) { $1 }
+            }
```

Comment:
This code took me a while to read it to understand it. 

Could we use extensions or a data structure to make it easier to read? Have code that looks more like this as a result:
```swift
            var deviceAttributes = defaultDeviceAttributes ?? [:] 
            if let customAttributes = customAttributes {
                deviceAttributes = deviceAttributes.merge(customAttributes)
            }
```

Reply:
Makes sense. Done. 👍